core/meta/src/TClass.cxx

```diff
@@ -4199,19 +4199,15 @@ TMethod *TClass::GetMethod(const char *method, const char *params,
 /// Find a method with decl id in this class or its bases.
 
 TMethod* TClass::FindClassOrBaseMethodWithId(DeclId_t declId) {
-   TFunction *f = GetMethodList()->Get(declId);
-   if (f) return (TMethod*)f;
+   if (auto method = GetMethodList()->Get(declId))
+      return reinterpret_cast<TMethod *>(method);
```

Comment:
'Thanks' to the auto keyword the validity of the reinterpret_cast becomes harder to Check.  (i.e. now one that to find out that GetMethodList returns a TListOfFunctions and then look up the return type of Get ... only then can one know that the cast is okay since TMethod inherits directly for the return type, TFunction).

Reply:
Well, it must be fine, since that's what the function returns. Using `reinterpret_cast` is better than just a C cast, as it will fail in more cases where the C cast would compile and give wrong results.

Reply:
> Well, it must be fine,

It is fine of course.  My point is about code readability (i.e. the ability for the human reader to verify the code :) ).

Reply:
I.e. in this case, because of the reinterpret_cast, I would use
```
if (TFunction *method = GetMethodList()->Get(declId))
      return reinterpret_cast<TMethod *>(method);
```